src/compile/projection/parse.ts

```diff
@@ -0,0 +1,65 @@
+import {SHAPE, X, Y} from '../../channel';
+import {Config} from '../../config';
+import {MAIN} from '../../data';
+import {isProjectionFieldDef} from '../../fielddef';
+import {GEOSHAPE} from '../../mark';
+import {Projection} from '../../projection';
+import {contains} from '../../util';
+import {VgProjection} from '../../vega.schema';
+import {Model} from '../model';
+import {UnitModel} from '../unit';
+import {ProjectionComponent} from './component';
+
+export function parseProjection(model: Model, config: Config) {
+  if (model instanceof UnitModel) {
+    model.component.projection = parseUnitProjection(model, config);
+  } else {
+    const mergedProjection = parseNonUnitProjections(model);
+  }
+}
+
+function parseUnitProjection(model: UnitModel, config: Config): ProjectionComponent {
+  const {projection, markDef, encoding} = model;
+
+    // TODO: make sure this actually works for both kinds of lookups, with GeoJSON transform, and with basic projection
+    const data = model.requestDataName(MAIN);
+
+    const width = model.getSizeSignalRef('width');
```

Comment:
I suggest using `{signal: model.getName('width')}` as I am about to restrict `getSizeSignalRef()` for using in assemble only.  

If you haven't, make sure that in `assemble`, you replace `size` signals with updated names.